### ISSUE TITLE
Scheduled Items Not Showing in Agenda

### ISSUE THREAD

**Issue by Anti-Distinctlyminty** [ihdavids/orgextended#10]
I did some research to find out how scheduled items are supposed to behave, so I'm assuming that [this explanation](https://emacs.stackexchange.com/questions/10504/understanding-scheduled-in-org-mode) is correct and that scheduled items should appear in the agenda without any kind of label on them

1. Add a new item
2. Hit alt+o, i, s and pick a datetime. The item is now scheduled.
3. Hit alt+o, o, a to open the agenda. The scheduled item does not show up.
4. Hit alt+o, t to add a TODO tag.
5. Again open the agenda. The scheduled item now appears.

You can see this happen in the following video (note there are items in the agenda from my 'in use' org file. I don't currently know how to make it not show these).
The link above also implies that plain timestamps should appear in the agenda, but they also do not show.

https://user-images.githubusercontent.com/1378483/107876847-f225a500-6ec8-11eb-8d2a-795ff56e85e5.mp4

**Comment by ihdavids** [comment]
Yup, you are right.

This is an area that I don't have full support. I haven't needed it, so I haven't added it yet.
If YOU are going to use this I will endeavor to extend my support. But I want to know you plan to use it, then I am happy to invest the time as I have time.

There are a couple of things here:

- Right now scheduled behaves like just adding an active date on a task, I don't continue to show the task until it is closed, I only show it on the scheduled date. I can make it support that, it was just simpler for me to have scheduled act like an active date when I built it, so I cheated at the time. NOTE: I don't even think I support active dates... I didn't have the distinction when I implemented it so I didn't bother adding the active dates. Again, all stuff I can fix if you want it.
- I only show tasks, not general headings with a schedule or active datetime. Again, I can support this if you want it.
- I don't show deadlines or respect them yet. Again, I usually just use scheduled for my personal needs, but I am happy to work to implement the full org here if you want it.
- Finally I do have respect for recurring dates (+1d etc) but it's not perfect.
- Habits do show, but they are missing some of the options that I don't use.

Things are even more complex. Org lets you define not only your task state flow (#+TODO: TODO INPROGRESS | DONE etc) with a variety of configuration mechanisms, people often define their own definitions of what a project is. Org has the concept of a property defining a project... I just use any task that has a task under it (IE the heading has a todo and the task has a todo) That is how the your life in plain text guy did it and I liked it and stuck with it. I figure someday I will get asked about that.... If people care, I can and will extend that stuff... It's just a bunch of work that I don't personally need....

If people are outspoken and ask. I am happy to invest the time where I can, or review PRs etc. I am just happy someone is getting some benefit from the system, I figured nobody would use it when I posted it, but it seems I was wrong, which is nice to know.

**Comment by Anti-Distinctlyminty** [comment]
I think that if that particular feature is there, then I will use it. However, I don't wish to overburden you either as I'm aware that this is a personal project which I'm abusing :D I suppose that as long as it's documented in either case, it should be alright.
It feels like Org is a huge monster that I have just awoken by treading on the tiniest part of its tail.

**Comment by ihdavids** [comment]
I am happy to work on it if I know it will help someone. Cant guarantee when. 

I can make use of it to. It just hasnt been a priority for me till now.

**Comment by ihdavids** [comment]
I just pushed 1.1.0
It has support for active timestamps and SCHEDULED is now more like the org docs. IE the item appears in the agenda until you close the task.

DEADLINE is not there yet, I'll will work on that when I have time.

**Comment by ihdavids** [comment]
You can revert a view back to the old behaviour by adding the parameter to the view: onlytasks in your configuration file.

**Comment by ihdavids** [comment]
(NOTE the default agenda view is a composite view made up of several views)

**Comment by Anti-Distinctlyminty** [comment]
Great work - Thank you!

**Comment by ihdavids** [comment]
1.1.1 now has basic DEADLINE support and fixes some issues with SCHEDULED.
Lots of stuff to improve there but the basics are in.

**Comment by Anti-Distinctlyminty** [comment]
Seems to have been a regression here. As of 1.1.5 scheduled items are no longer appearing in the day agenda. Nothing strange  and no errors in the Sublime console.


https://user-images.githubusercontent.com/1378483/108327992-5c419100-71cc-11eb-97fc-ce811228553c.mp4

**Comment by ihdavids** [comment]
Huh, I will take a look as soon as I can.
That does seem broken. 

I did make a bunch of changes that were supposed to help with dates vs dates and times. (I stole orgparse a python orgmode parser that I am using and they split dates and datetimes into separate disctinct objects and it's been nothing but headache in the agenda.) I am not seeing it locally yet, but I think I haven't updated to 1.1.6 yet today.

**Comment by ihdavids** [comment]
Hmmm, have you tried with 1.1.6? I did fix a major regression last night that I introduced accidentally with some of the date vs date time stuff. I am not able to duplicate your example above. :(

Sorry about the frustration, hopefully I am done mucking about now.
I decided when I started supporting DEADLINE that I needed to support

DEADLINE: <blah -3d>  -- Warning tokens
<blah blahtime +1d> -- recurrence tokens

I royally broke things as I worked through those.

**Comment by Anti-Distinctlyminty** [comment]
Yeah, I'm on 1.16
It's weird. I'm adding SCHEDLED items now and they are showing up. I wish I was able to offer more assistance, but I'm so unfamiliar with org and the way it works. Less unfamiliar than previously, of course, but still.

One thing that I've been meaning to ask - does the plugin keep a central store of nodes every time a file is opened? Does it remember all org files that it has ever opened?
Basically, how does the agenda get it's data? And does that data need to be regenerated when the plugin is updated?

**Comment by ihdavids** [comment]
Huh. Odd.

So the plugin scans all files in your orgdir (minus exclusions ) and files you have opened. It does this in a background thread at startup for files you dont currently have opened and then as a file is saved or you access an org feature that needs the db the file gets rescanned. The agenda just walks the db and shows you everything in the db. I am wondering if this is a save file issue now that you mention it. We have already ascertained I save files much more often than you. ;)

**Comment by ihdavids** [comment]
The setup section mentions the ordirs option but doesnt do a good job of explaining that.

**Comment by ihdavids** [comment]
Are you flipping back to the aganda tab? Note the agenda wont update unless you regenrate it. It is static and doesnt update automatically.


Hmm trying to think what else could be going on.

I will add some logging tonight and get you to turn that on so we can see if the agenda generation is seeing your new items at all

**Comment by ihdavids** [comment]
Okay, I am at a loss. I cannot duplicate what you are seeing. 
I can make things not appear if I don't save the file... 
The odd thing is why the week view is able to see these items. It is working off the same DB...

**Comment by ihdavids** [comment]
![need_to_save](https://user-images.githubusercontent.com/1059210/108453777-03034d00-7220-11eb-96c6-e5a6b89793aa.gif)

**Comment by ihdavids** [comment]
![need_to_save2](https://user-images.githubusercontent.com/1059210/108453794-0b5b8800-7220-11eb-9cb7-310c23a0e27a.gif)

**Comment by ihdavids** [comment]
Okay, I think I got it to do what is happening to you.
The issue is saving I think.

Notice in the second one I fold the heading... When you fold and the current buffer is marked as modified since the last "parse" I dynamically reparse the file in memory (from the buffer) Which updated the DB so the item can appear in the agenda....

When you just wrote in the buffer, the agenda was not updated. I try to not suck up cycles, so I only reparse when I absolutely have to like when you are doing something that is context aware in the buffer (like folding) and the file is modified OR you are saving the file.

I am not sure when I will get around to fixing this one. I would need to validate all open buffers to ensure they don't need to be reparsed when regenerating the agenda. I can do that, but sometimes people have a massive number of tabs open.... I would need to do some testing to ensure I don't cause goofy hitches are issues there.....

**Comment by ihdavids** [comment]
BUT, I will add this to my todo list.
